client/state/themes/selectors.js

```diff
@@ -357,7 +357,7 @@ export function getThemeDetailsUrl( state, theme, siteId ) {
  * @return {?String}        Theme setup instructions URL
  */
 export function getThemeSupportUrl( state, theme, siteId ) {
-	if ( ! theme || ! isPremiumTheme( theme ) ) {
+	if ( isJetpackSite( state, siteId ) || ! theme || ! isThemePremium( state, theme.id ) ) {
```

Comment:
Thinking about it, this is trickier than it seems 😄 

The `getThemeSupportUrl` returns the URL of the theme sheet's 'Setup' tab, which displays the text contained in the theme object's `supportDocumentation` attr, which JP sites don't return. (But as discussed yesterday, we might query WP.com for the theme and merge attributes from it, much like we do for WP.org already.)

Finally, keep in mind that we plan to filter themes that were installed from WP.com from the 'Local Theme' list, so it'll be only in the WP.com themes list below, where I think we'll set the `siteId` prop to `false` to signal we're looking for the generic list (not quite sure about this yet).

For now, I'll try it without the `isJetpackSite` check locally and see if it points me to an empty 'Setup' tab. If it doesn't, I'll remove the check 😄